packages/contracts-bedrock/contracts/universal/StandardBridge.sol

```diff
@@ -416,7 +418,7 @@ abstract contract StandardBridge {
 
             OptimismMintableERC20(_localToken).burn(_from, _amount);
         } else {
-            IERC20(_localToken).safeTransferFrom(_from, address(this), _amount);
+            IERC20(_localToken).transferFrom2(_from, address(this), _amount);
```

Comment:
would be nice to optionally be able to pack in a permit signature so users can get 1-tx flow. I.e. can do 

```
IERC20(_localToken).permit2(_from, address(this), _amount, deadline, ...sig)
IERC20(_localToken).transferFrom2(...)
```

This way user can sign permit and just include it with their bridge tx. Would require breaking the external interface or adding a separate entrypoint unless you want to do something wonky like pack it in extraData

Reply:
I definitely agree that having a 1 tx flow is where its at. Haven't decided whether the extradata approach or alternative entrypoint is a better approach. The alternate entrypoint requires some refactoring or code duplication, and I didn't want to introduce that into the codebase quite yet.

The extradata is not used for anything yet, was thinking of the possibility of making the first byte of extradata an enum and then abi decoding based on that. Then its extensible into the future. Another idea for the extradata would be including a meta tx to pay out to a relayer that finalizes a withdrawal, but thats even further out into the future